tests/ci/kickoff_devicefarm_job.sh

```diff
@@ -0,0 +1,170 @@
+  fi
+  if [[ -z "${DEVICEFARM_PROJECT+x}" || -z "${DEVICEFARM_PROJECT}" ]]; then
+    export DEVICEFARM_PROJECT='arn:aws:devicefarm:us-west-2:620771051181:project:d1e78543-a776-49c5-9452-9a2b3448b728'
+  fi
+  if [[ -z "${FIPS+x}" || -z "${FIPS}" ]]; then
+    export FIPS=false
+  fi
+  if [[ -z "${RELEASE+x}" || -z "${RELEASE}" ]]; then
+    export RELEASE=false
+  fi
+  if [[ -z "${SHARED+x}" || -z "${SHARED}" ]]; then
+    export SHARED=false
+  fi
+  if [[ -z "${DEVICEFARM_DEVICE_POOL+x}" || -z "${DEVICEFARM_DEVICE_POOL}" ]]; then
+    if [[ "${FIPS}" = true ]]; then
+      # Device pool arn for FIPS.
+      export DEVICEFARM_DEVICE_POOL='arn:aws:devicefarm:us-west-2:620771051181:devicepool:d1e78543-a776-49c5-9452-9a2b3448b728/4726586b-cdbc-4dc0-98a5-38e7448e3691'
+    else
+      # Device pool arn for non-FIPS.
+      export DEVICEFARM_DEVICE_POOL='arn:aws:devicefarm:us-west-2:620771051181:devicepool:d1e78543-a776-49c5-9452-9a2b3448b728/4e72604c-86eb-41b6-9383-7797c04328b4'
+    fi
+  fi
+  if [[ -z "${AWS_REGION+x}" || -z "${AWS_REGION}" ]]; then
+    export AWS_REGION='us-west-2'
+  fi
+}
+
+function compile_for_android() {
+  # |ANDROID_APK| and |ANDROID_TEST_APK| are apk names corresponding to the settings in the test runner gradle file.
+  # The working directory of this script should under `tests/ci`. 
+  # AWSLCAndroidTestRunner should be at `tests/ci/android/AWSLCAndroidTestRunner`.
+  cd android/AWSLCAndroidTestRunner
+  export ANDROID_APK_LOCATION='android/AWSLCAndroidTestRunner/app/build/outputs/apk'
+  if [[ "${FIPS}" = true ]]; then
+    ./gradlew assembleDebug assembleAndroidTest -PFIPS
+    export ANDROID_APK="${ANDROID_APK_LOCATION}/debug/awslc_fips.apk"
+    export ANDROID_TEST_APK="${ANDROID_APK_LOCATION}/androidTest/debug/awslc_fips-androidTest.apk"
+  else
+    if [[ "${RELEASE}" = true ]]; then
+      if [[ "${SHARED}" = true ]]; then
+        ./gradlew assembleDebug assembleAndroidTest -PRelease -PShared
+        export ANDROID_APK="${ANDROID_APK_LOCATION}/debug/awslc_shared_rel.apk"
+        export ANDROID_TEST_APK="${ANDROID_APK_LOCATION}/androidTest/debug/awslc_shared_rel-androidTest.apk"
+      else
+        ./gradlew assembleDebug assembleAndroidTest -PRelease
+        export ANDROID_APK="${ANDROID_APK_LOCATION}/debug/awslc_static_rel.apk"
+        export ANDROID_TEST_APK="${ANDROID_APK_LOCATION}/androidTest/debug/awslc_static_rel-androidTest.apk"
+      fi
+    else
+      if [[ "${SHARED}" = true ]]; then
+        ./gradlew assembleDebug assembleAndroidTest -PShared
+        export ANDROID_APK="${ANDROID_APK_LOCATION}/debug/awslc_shared_dbg.apk"
+        export ANDROID_TEST_APK="${ANDROID_APK_LOCATION}/androidTest/debug/awslc_shared_dbg-androidTest.apk"
+      else
+        ./gradlew assembleDebug assembleAndroidTest
+        export ANDROID_APK="${ANDROID_APK_LOCATION}/debug/awslc_static_dbg.apk"
+        export ANDROID_TEST_APK="${ANDROID_APK_LOCATION}/androidTest/debug/awslc_static_dbg-androidTest.apk"
```

Comment:
Maybe comment the state for each block for increased readability i.e. 
\# Release Shared
\# Release Static
\# Debug Shared
\# Debug Static